test/specs/validate-spec/invalid/3.x/duplicate-header-params.yaml

```diff
@@ -0,0 +1,31 @@
+openapi: 3.0.0
+info:
+  version: "1.0.0"
+  title: Invalid API
+
+paths:
+  /users/{username}:
+    parameters:
+      - name: username
+        in: path
+        required: true
+        schema:
+          type: string
+      - name: foo           # <---- Duplicate param
+        in: header
+        required: false
+        schema:
+          type: string
+      - name: username
+        in: header
+        schema:
+          type: string
+      - name: foo           # <---- Duplicate param
```

Comment:
This library has long had validation support for this kind of thing for Swagger 2.0 specs, this work will extend that to OAS 3.0+.